feat(python): add sql() and show() convenience method to QueryBuilder

DevChrisCross opened the pull request:
# Description
Adds user friendly methods, namely sql() and show() to the QueryBuilder

# Related Issue(s)
- closes #3076

# Documentation

<!---
Share links to useful documentation
--->

ion-elgreco commented:
@DevChrisCross the .SQL() should rather give an object that's lazy executed, similar to duckdb's python API: 


https://duckdb.org/docs/api/python/reference/

You can see the method .SQL() and .show() there

djouallah commented:
for what is worth it,  instead of qb.show().execute(query)
  it should be
` qb.execute(query).show()`

DevChrisCross commented:
I've introduced a QueryResult class to accommodate the lazy execution, and in turn, introduced fetchall() as well, let me know if it is okay @ion-elgreco @djouallah

djouallah commented:
do you mind adding   qb.sql(query).show() too please

DevChrisCross commented:
> do you mind adding qb.sql(query).show() too please

Ah yes, that was already added as well :)

ion-elgreco commented:
@DevChrisCross looks good, perhaps also add a  to_arrow() which returns a pyarrow table and to_pandas() on it as well : )

rtyler commented:
I am squashing this and will then merge, I love seeing more improvements here :smile:

DevChrisCross commented:
@rtyler Oh it was merged already? I was initially trying to accommodate comments @ion-elgreco regarding the to_arrow and to_pandas methods 😅 Is it still needed?

rtyler commented:
@DevChrisCross I think those are good ideas to add in additional pull requests :smile:
